internal/beater/config/sampling.go

```diff
@@ -52,6 +52,7 @@ type TailSamplingConfig struct {
 	TTL                   time.Duration         `config:"ttl" validate:"min=1s"`
 	StorageLimit          string                `config:"storage_limit"`
 	StorageLimitParsed    uint64
+	DiscardOnWriteFailure bool `config:"discard_on_write_failure"`
```

Comment:
[to reviewer] Using `discard_on_write_failure` instead of `index_on_write_failure` to have a default of false, and give a stronger sense of opt-in to data loss. Lmk if there's a better name, e.g. `drop_on_write_failure`.